pkg/network/link/discovery.go

```diff
@@ -45,6 +45,28 @@ func DiscoverByNetwork(handler driver.NetworkHandler, networks []v1.Network, sub
 	return linkByNames(handler, ifaceNames)
 }
 
+func DiscoverByDefaultGateway(family int) ([]netlink.Link, error) {
+	routes, err := netlink.RouteList(nil, family)
+	if err != nil {
+		return nil, fmt.Errorf("failed looking for default gw routes: %w", err)
+	}
+	selectedLinksMap := map[string]netlink.Link{}
+	for _, r := range routes {
+		if r.Dst == nil {
```

Comment:
Asking Just to understand, having a route with no destination is the best indication for it to be the default GW route?

Reply:
AFAIU it can also be explicitly defined: https://github.com/vishvananda/netlink/blob/d13535d71ed3e430e26c7538cd6266d026b26f54/route_test.go#L84

We may need to find a better source of truth.